docs/tutorials/concepts/2dmotionprofiling.md

```diff
@@ -0,0 +1,130 @@
+2D Motion Profiling
+===================
+
+Motion Profiling is a controls algorithm that defines a movement as a
+series of steps. In one dimension, it defines a motor's position,
+speed, acceleration, etc. at every timestep during the movement. In two
+dimensions, it defines your robot's position, speed, acceleration, etc.
+and heading at every timestep during the movement.
+
+There is a lot of interesting math that goes into calculating these
+values, which you can learn about in examples like these:
+
+-   <http://www2.informatik.uni-freiburg.de/~lau/students/Sprunk2008.pdf>
+-   <https://github.com/AtsushiSakai/PythonRobotics/blob/master/PathPlanning/QuinticPolynomialsPlanner/animation.gif>
+-   <https://www.dis.uniroma1.it/~deluca/rob1_en/13_TrajectoryPlanningJoints.pdf>
+
+OkapiLib, through
+[Pathfinder](https://github.com/JacisNonsense/Pathfinder), does this
+math for you and ties both the motion profiling and profile following
+actions to the same [async
+controllers](../walkthrough/autonomous-movement-async.html) that are
+used for other simpler movements. Because OkapiLib's motion profile
+generation uses
+[Pathfinder](https://github.com/JacisNonsense/Pathfinder), there are a
+few open issues you should know about:
+
+-   Pathfinder cannot generate negative velocities, so backward movements and very tight turns do not work
+  -   Moving backwards:
+            <https://github.com/JacisNonsense/Pathfinder/issues/39>
+  -   Tight turns:
+            <https://github.com/JacisNonsense/Pathfinder/issues/38>
+
+-   Very long movements (typically movements much longer than a VEX field) can potentially never reach maximum speed
+       -   <https://github.com/JacisNonsense/Pathfinder/issues/43>
+
+First, let's initialize a Chassis Controller to pass into the factory
+method for the motion profile Async controller. Following that, you can
+create a profile Async Controller.
+
+
+
+> The \"linear\" properties of the chassis given to
+> `AsyncControllerFactory::motionProfile` are for its forward/backwards
```

Comment:
API docs can be referenced like `[waitUntilSettled()](@ref okapi::ChassisController::waitUntilSettled)`.